File: Desktop/DSA/T73/intersection.py
1# Problem statement
# You are given two arrays 'A' and 'B' of size 'N' and 'M' respectively. Both these arrays are sorted in non-decreasing order. You have to find the intersection of these two arrays.

# Intersection of two arrays is an array that consists of all the common elements occurring in both arrays.

# Note :
# 1. The length of each array is greater than zero.
# 2. Both the arrays are sorted in non-decreasing order.
# 3. The output should be in the order of elements that occur in the o

def intersection(arr1:list[int],n:int,arr2:list[int],m:int)->list[int]:
    
    i=0
    j=0
    result=[]
    
    while i<n  and j<m:
        
        if arr1[i] == arr2[j]:
            
            result.append(arr1[i])
            
            i +=1
            j +=1
            
        elif arr1[i] < arr2[j]:
            
            i +=1
            
        else:
            
            j += 1
            
    return result

arr = [1, 2, 4, 5, 6]
brr = [2, 3, 5, 7]
n = len(arr)
m = len(brr)

print(intersection(arr,n,brr,m))

File: Documents/ANUDIP JAVA/Intersection.java

import java.util.ArrayList;

public class Intersection {
	
	public static int[] common(int[] arr1 , int[]arr2) {
		
		ArrayList<Integer> store = new ArrayList<>();
		
		for (int i = 0; i <arr1.length ; i++) {
			
			for (int j = 0; j <arr2.length ; j++) {
				
				if (arr1[i] == arr2[j] && !store.contains(arr1[i])) {
					
					store.add(arr1[i]);
					break;
				}
			}
		}
		
        int[] result = new int[store.size()];
        for (int i = 0; i < store.size();i++ ) {
            result[i] = store.get(i);
        }
        return result;
		
	}
	
	  public static void main(String[] args) {
	        int[] arr1 = {1, 2, 2, 3, 4};
	        int[] arr2 = {2, 3, 3, 5};

	        int[] intersection = common(arr1, arr2);

	        System.out.println("Array 1: " + java.util.Arrays.toString(arr1));
	        System.out.println("Array 2: " + java.util.Arrays.toString(arr2));
	        System.out.println("Intersection: " + java.util.Arrays.toString(intersection));
	    }

}
